gateway/platforms/servlet/src/main/java/io/apiman/gateway/platforms/servlet/connectors/ok/HttpURLConnectionImpl.java

```diff
@@ -256,9 +255,11 @@ private static String responseSourceHeader(Response response) {
     return sink.outputStream();
   }
 
-  @Override public final Permission getPermission() throws IOException {
+  @Override public final Permission getPermission() {
     String hostName = getURL().getHost();
-    int hostPort = Util.getEffectivePort(getURL());
+    int hostPort = url.getPort() != -1
+          ? url.getPort()
+          : HttpUrl.defaultPort(url.getProtocol());
```

Comment:
Nice.

Reply:
Was from upstream, can't take credit for that one 😂